exchanges/binanceus/binanceus_test.go

```diff
@@ -0,0 +1,1655 @@
+		Description: "WITHDRAW IT ALL",
+		Crypto: withdraw.CryptoRequest{
+			Address: core.BitcoinDonationAddress,
+			Chain:   "BSC",
+		},
+	}
+
+	_, err := bi.WithdrawCryptocurrencyFunds(context.Background(),
+		&withdrawCryptoRequest)
+
+	switch {
+	case areTestAPIKeysSet() && err != nil:
+		if strings.Contains(err.Error(), "amount must be greater than zero") {
+			return
+		}
+		t.Error("Binanceus Withdraw() error", err)
+	case !areTestAPIKeysSet() && err == nil:
+		t.Error("Binanceus Withdraw() expecting an error when no keys are set")
+	}
+}
+
+func TestGetFee(t *testing.T) {
+	t.Parallel()
+
+	var feeBuilder = &exchange.FeeBuilder{
+		Amount:        1,
+		FeeType:       exchange.CryptocurrencyTradeFee,
+		Pair:          currency.NewPair(currency.BTC, currency.LTC),
+		PurchasePrice: 1,
+	}
+	_, er := bi.GetFeeByType(context.Background(), feeBuilder)
+	if er != nil {
+		t.Fatal("Binanceus GetFeeByType() error", er)
+	}
+	var withdrawalFeeBuilder = &exchange.FeeBuilder{
+		Amount:        1,
+		FeeType:       exchange.CryptocurrencyWithdrawalFee,
+		Pair:          currency.NewPair(currency.BTC, currency.LTC),
+		PurchasePrice: 1,
+	}
+	_, er = bi.GetFeeByType(context.Background(), withdrawalFeeBuilder)
+	if er != nil {
+		t.Fatal("Binanceus GetFeeByType() error", er)
+	}
+	var offlineFeeTradeBuilder = &exchange.FeeBuilder{
+		Amount:        1,
+		FeeType:       exchange.OfflineTradeFee,
+		Pair:          currency.NewPair(currency.BTC, currency.LTC),
+		PurchasePrice: 1,
+	}
+	_, er = bi.GetFeeByType(context.Background(), offlineFeeTradeBuilder)
+	if er != nil {
+		t.Fatal("Binanceus GetFeeByType() error", er)
+	}
+}
+
+func TestGetHistoricCandles(t *testing.T) {
+	t.Parallel()
+	pair := currency.NewPair(currency.BTC, currency.USDT)
+	startTime := time.Date(2020, 9, 1, 0, 0, 0, 0, time.UTC)
+	endTime := time.Date(2021, 2, 15, 0, 0, 0, 0, time.UTC)
+	_, er := bi.GetHistoricCandles(context.Background(), pair, asset.Spot, startTime, endTime, kline.Interval(time.Hour*5))
+	if !strings.Contains(er.Error(), "interval not supported") {
+		t.Errorf("Binanceus GetHistoricCandles() expected %s, but found %v", "interval not supported", er)
+	}
+	_, er = bi.GetHistoricCandles(context.Background(), pair, asset.Spot, time.Time{}, time.Time{}, kline.Interval(time.Hour*4))
+	if er != nil {
+		t.Error("Binanceus GetHistoricCandles() error", er)
+	}
+}
+
+func TestGetHistoricCandlesExtended(t *testing.T) {
+	t.Parallel()
+	pair := currency.NewPair(currency.BTC, currency.USDT)
+	startTime := time.Date(2020, 9, 1, 0, 0, 0, 0, time.UTC)
+	endTime := time.Date(2021, 2, 15, 0, 0, 0, 0, time.UTC)
+	_, er := bi.GetHistoricCandlesExtended(context.Background(), pair, asset.Spot, startTime, endTime, kline.Interval(time.Hour*5))
+	if !strings.Contains(er.Error(), "interval not supported") {
+		t.Errorf("Binanceus GetHistoricCandlesExtended() expected %s, but found %v", "interval not supported", er)
+	}
+	startTime = time.Unix(time.Now().Unix()-int64(time.Hour*30), 0)
```

Comment:
This was causing intensive issues as the start date was being set as `-3420362-09-24 15:33:24 +1004` and there are many 4 hour intervals between the year -3420362 and 2022
```suggestion
	startTime = time.Now().Add(-time.Hour *30)
```